deploy/k8s/env.sh

```diff
@@ -11,14 +11,24 @@ hostname="$(kubectl get service ingress-nginx-controller -o jsonpath='{.status.l
 port="$(kubectl get service ingress-nginx-controller -o jsonpath='{.spec.ports[?(@.appProtocol=="http")].port}')"
 
 # Get password for system/admin user generate by `helm install`
-NUODB_CP_URL_BASE="http://$hostname:$port/nuodb-cp"
-NUODB_CP_USER=system/admin
-NUODB_CP_PASSWORD="$(kubectl get secret dbaas-user-system-admin -o jsonpath='{.data.password}' | base64 -d)"
+export NUODB_CP_URL_BASE="http://$hostname:$port/nuodb-cp"
+export NUODB_CP_USER=system/admin
+export NUODB_CP_PASSWORD="$(kubectl get secret dbaas-user-system-admin -o jsonpath='{.data.password}' | base64 -d)"
+
+# Use token authentication if server is configured to generate tokens
+if [ -n "$(kubectl get secrets nuodb-cp-runtime-config -o jsonpath='{.data.secretPassword}' --ignore-not-found)" ]; then
+    NUODB_CP_TOKEN="$(nuodb-cp httpclient POST login --jsonpath token --unquote)"
+    if [ -n "$NUODB_CP_TOKEN" ]; then
+        unset NUODB_CP_USER
+        unset NUODB_CP_PASSWORD
+    fi
+fi
```

Comment:
Once we bump the minimum supported CP version to a version that supports token auth, are we going to lose automated test coverage of login with basic credentials?

Reply:
This is one of the 3 test drivers. KWOK and External will still use basic auth.